### PR TITLE
Test automatic pom with dependencies generator for aar

### PR DESCRIPTION
**Pull request by originx**
I tested this and it works very well, we just define artifactID, groupName and version dependencies are generated automatically in pom file so the end use can only do this:

 `implementation 'com.github.hannesa2:DbPreferences:feature~testPomGenerator-SNAPSHOT'`
vs

`
    implementation 'com.github.hannesa2:DbPreferences:feature~testPomGenerator-SNAPSHOT'
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
    implementation "org.reactivestreams:reactive-streams:1.0.2"
    implementation "io.reactivex.rxjava2:rxjava:2.2.2"
    implementation 'com.google.code.gson:gson:2.8.5'
    // room
    implementation "com.commonsware.cwac:saferoom:0.4.4"
    implementation "android.arch.persistence.room:runtime:$room"
    kapt "android.arch.persistence.room:compiler:$roomruntime"
    implementation "android.arch.persistence.room:rxjava2:$room"
    implementation "android.arch.persistence.room:testing:$room"`

### CODE REVIEW

File: DBPrefs/build.gradle [hunk 1]
```diff
@@ -1,7 +1,8 @@
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 apply plugin: 'kotlin-kapt'
-
+apply plugin: 'digital.wup.android-maven-publish'
```
Comment (hannesa2) [inline DBPrefs/build.gradle]: Till now, we didn't use it

File: DBPrefs/build.gradle [hunk 2]
```diff
@@ -22,6 +23,17 @@ android {
     }
 }
 
+publishing {
+    publications {
+        mavenAar(MavenPublication) {
```
Comment (hannesa2) [inline DBPrefs/build.gradle]: this is the magic, which include everything in pom, to just use one single `implements` 
Am I right ?

File: DBPrefs/build.gradle (hunk 1 above)
Comment (originx) [reply DBPrefs/build.gradle]: thats the plugin that generates poms for aar

Comment (originx) [reply DBPrefs/build.gradle]: it can also upload to maven repository if you specify it

File: DBPrefs/build.gradle (hunk 2 above)
Comment (originx) [reply DBPrefs/build.gradle]: yep

**Pull request merged**